pyiron/atomistics/master/elastic.py

```diff
@@ -0,0 +1,271 @@
+
+    @staticmethod
+    def modify_job(job, parameter):
+        job.structure = parameter[1]
+        return job
+
+# ToDo: not all abstract methods implemented
+class ElasticTensor(AtomisticParallelMaster):
+    """
+    Class to calculate the elastic tensor and isotropic elastic constants
+
+    Example:
+
+    >>> job = pr.create_job('SomeAtomisticJob', 'atomistic')
+    >>> job.structure = pr.create_structure('Fe', 'bcc', 2.83)
+    >>> elastic = job.create_job('ElasticTensor', 'elastic')
+    >>> elastic.run()
+
+    The input parameters might not be chosen adequately. If you have a large
+    computation power, increase `min_num_measurements`. At the same time, make
+    sure to choose an orientation which maximizes the number symmetry operations.
+    Also if the child job does not support pressure, better increase the number
+    of measurements.
+    """
+    def __init__(self, project, job_name="elastic"):
+        """
+
+        Args:
+            project:
+            job_name:
+        """
+        super().__init__(project, job_name)
+        self.__name__ = "ElasticTensor"
+        self.__version__ = "0.1.0"
+
+        self.input["min_num_measurements"] = (11, "minimum number of samples to be taken")
+        self.input["min_num_points"] = (105, "minimum number of data points"
+            + "(number of measurements will be min_num_points/len(rotations))")
+        self.input["max_strain"] = (
+            0.01,
+            "relative volume variation around volume defined by ref_ham",
+        )
+        self.input['strain_matrices'] = []
+        self.input['use_symmetry'] = True
+        self.input['rotations'] = []
+        self.input['use_elements'] = (True, 'whether or not consider chemical elements for '
+                                            + 'the symmetry analysis. Could be useful for SQS')
+        self._job_generator = ElasticJobGenerator(self)
+
+    @property
+    def _number_of_measurements(self):
+        return max(
+            self.input['min_num_measurements'],
+            int(self.input['min_num_points']/max(len(self.input['rotations']), 1))
+        )
+
+    def validate_ready_to_run(self):
+        super().validate_ready_to_run()
+        if self.input['use_symmetry'] and len(self.input['rotations'])==0:
+            rotations = self.ref_job.structure.get_symmetry(use_elements=self.input['use_elements'])['rotations']
+            _, indices = np.unique(np.round(rotations, 6), axis=0, return_inverse=True)
+            rotations = rotations[np.unique(indices)]
+            self.input['rotations'] = rotations.tolist()
+        if len(self.input['strain_matrices'])==0:
+            eps_lst = np.random.random((int(self._number_of_measurements), 3, 3))-0.5
+            eps_lst *= self.input['max_strain']
+            eps_lst += np.einsum('nij->nji', eps_lst)
+            self.input['strain_matrices'] = eps_lst.tolist()
+
+    def list_structures(self):
```

Comment:
Since this is potentially called by users a very brief docstring would be nice.  Just so it's more visible on readthedocs.

Reply:
Actually this one comes from `AtomisticParallelMaster` and I guess the function name must be like this. Maybe @jan-janssen can say more about it.

Reply:
Sorry I was wrong - it does nothing. I just deleted it

Reply:
And I guess it's a useful function, so I put it back with a DocString :D

Reply:
... and I just realized that it cannot work in the current framework so in the end I removed it again :D